Buildings/Fluid/CHPs/BaseClasses/EngineConVol.mo

```diff
@@ -0,0 +1,116 @@
+within Buildings.Fluid.CHPs.BaseClasses;
+model EngineConVol "Heat exchange within the engine control volume"
+  extends Modelica.Blocks.Icons.Block;
+
+  replaceable parameter Buildings.Fluid.CHPs.Data.Generic per
+    "Performance data"
+    annotation (Placement(transformation(extent={{60,60},{80,80}})));
+
+  parameter Modelica.SIunits.Temperature TEngIni "Initial engine temperature";
+
+  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a TRoo
+    "Heat port for room temperature"
+    annotation (Placement(transformation(extent={{-110,50},{-90,70}}),
+      iconTransformation(extent={{-110,48},{-90,68}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput QGen(
+    final unit="W") "Heat generation within the engine"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+      iconTransformation(extent={{-140,-20},{-100,20}})));
+  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a TWat
+    "Water volume temperature"
+    annotation (Placement(transformation(extent={{-110,-70},{-90,-50}}),
+      iconTransformation(extent={{-110,-70},{-90,-50}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput TEng(
+    final unit="K",
+    final quantity="ThermodynamicTemperature") "Engine temperature"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),
+      iconTransformation(extent={{100,-20},{140,20}})));
+
+protected
+  constant Modelica.SIunits.Density rhoWat=1000 "Water density";
+  constant Modelica.SIunits.SpecificHeatCapacity cWat=4180 "Water specific heat";
+  Modelica.Thermal.HeatTransfer.Components.ThermalConductor UAhx(
```

Comment:
It is surprising that the report does not recommend a correlation of the UA value to the water mass flow rate!